ff-rewrite-typescript-id-allocator/src/utilities.ts

```diff
@@ -139,3 +141,58 @@ export function uuidStringFromBytes(uuidBytes: Uint8Array | undefined): string |
 	}
 	return uuidString;
 }
+
+// xxxxxxxx-xxxx-Mxxx-Nxxx-xxxxxxxxxxxx
+const versionMask = 0x4n << (19n * 4n); // Version 4
+const variantMask = 0x8n << (15n * 4n); // Variant RFC4122 (1 0 x x)
+const upperMask = 0xffffffffffffn << (20n * 4n);
+// Upper mask when version/variant bits are removed
+const strippedUpperMask = upperMask >> 6n;
+const middieBittiesMask = 0xfffn << (16n * 4n);
+// Middie mask when version/variant bits are removed
+const strippedMiddieBittiesMask = middieBittiesMask >> 2n;
+// Note: leading character should be 3 to mask at 0011
+// The more-significant half of the N nibble is used to denote the variant (10xx)
+const lowerMask = 0x3fffffffffffffffn;
+// Used to help with stringifying bigints which would otherwise drop trailing zeros
+const precisionMask = 0x1n << 128n;
+
+export function numericUuidFromStableId(stableId: StableId): NumericUuid {
+	const uuidU128 = BigInt(`0x${stableId.replace(/-/g, "")}`);
```

Comment:
What is this doing?

Reply:
it removes all occurrences of '-' from the input string